src/Activation/NotificationPermissions.spec.tsx

```diff
@@ -0,0 +1,134 @@
+        <OnboardingContext.Provider value={onboardingProviderValue}>
+          <PermissionsContext.Provider value={permissionsProviderValue}>
+            <NotificationPermissions />
+          </PermissionsContext.Provider>
+        </OnboardingContext.Provider>,
+      )
+
+      fireEvent.press(getByLabelText("Enable Notifications"))
+
+      expect(notificationRequestSpy).toHaveBeenCalled()
+    })
+    it("marks onboarding complete", async () => {
+      const completeOnboardingSpy = jest.fn()
+      const onboardingProviderValue = createOnboardingProviderValue(
+        completeOnboardingSpy,
+      )
+
+      const notificationRequestSpy = jest.fn()
+      const permissionsProviderValue = createPermissionProviderValue(
+        notificationRequestSpy,
+      )
+
+      const { getByLabelText } = render(
+        <OnboardingContext.Provider value={onboardingProviderValue}>
+          <PermissionsContext.Provider value={permissionsProviderValue}>
+            <NotificationPermissions />
+          </PermissionsContext.Provider>
+        </OnboardingContext.Provider>,
+      )
+
+      fireEvent.press(getByLabelText("Enable Notifications"))
+
+      await waitFor(() => {
+        expect(completeOnboardingSpy).toHaveBeenCalled()
+      })
+    })
+  })
+  describe("when a user does not enable notifications", () => {
+    it("does not request permissions", () => {
+      const onboardingProviderValue = createOnboardingProviderValue(jest.fn())
+
+      const notificationRequestSpy = jest.fn()
+      const permissionsProviderValue = createPermissionProviderValue(
+        notificationRequestSpy,
+      )
+
+      const { getByText } = render(
+        <OnboardingContext.Provider value={onboardingProviderValue}>
+          <PermissionsContext.Provider value={permissionsProviderValue}>
+            <NotificationPermissions />
+          </PermissionsContext.Provider>
+        </OnboardingContext.Provider>,
+      )
+
+      fireEvent.press(getByText("Maybe later"))
+
+      expect(notificationRequestSpy).not.toHaveBeenCalled()
+    })
+    it("marks onboarding complete", async () => {
+      const completeOnboardingSpy = jest.fn()
+      const onboardingProviderValue = createOnboardingProviderValue(
+        completeOnboardingSpy,
+      )
+
+      const notificationRequestSpy = jest.fn()
+      const permissionsProviderValue = createPermissionProviderValue(
+        notificationRequestSpy,
+      )
+
+      const { getByText } = render(
+        <OnboardingContext.Provider value={onboardingProviderValue}>
+          <PermissionsContext.Provider value={permissionsProviderValue}>
+            <NotificationPermissions />
+          </PermissionsContext.Provider>
+        </OnboardingContext.Provider>,
+      )
+
+      fireEvent.press(getByText("Maybe later"))
+
+      await waitFor(() => {
+        expect(completeOnboardingSpy).toHaveBeenCalled()
+      })
+    })
+  })
+})
+
+const createPermissionProviderValue = (
```

Comment:
Maybe is just the pattern but don't we have these helpers in a few places, could we benefit from having a factory for these? Please ignore if there are just a couple of places where it makes sense to test the permissions context

Reply:
Yeah that sounds good! I'll pick that up in a future PR.